Erwinspeybroeck

erwinspeybroeck opened the pull request:
This PR resolves https://github.com/adewg/ICAR/issues/184

First version of the statistics resource

erwinspeybroeck committed:
added statitics

Statistics API is added to the managment section

erwinspeybroeck committed:
Update icarStatisticsResource.json

array's were not well defined

erwinspeybroeck committed:
statistics parameters

parameters changed for better use in combination with statistics

cookeac commented on types/icarGroupType.json:
```diff
@@ -0,0 +1,22 @@
+    "icarGroupType": {
+      "$ref": "../enums/icarGroupType.json"
+    },
+    "denombinator": { 
```
typo - denominator

erwinspeybroeck committed:
updated based on review

metric -> scheme instead of enum
enums in camelcase
type denominator

erwinspeybroeck commented:
I changed the status of this pull request from "Draft" to "Ready for preview"

AndreasSchultzGEA commented on enums/icarStatisticsPurposeType.json:
```diff
@@ -0,0 +1,15 @@
+
+  "enum": [
+    "TestDay",
+    "feeding",
```
also to uppercase

AndreasSchultzGEA commented on enums/icarGroupType.json:
```diff
@@ -0,0 +1,13 @@
+{
```
I'm a little bit confused, as the IcarGroupType is defined at enums as well as under types.

AndreasSchultzGEA commented on url-schemes/managementURLScheme.json:
```diff
@@ -442,6 +442,48 @@
+            "get": {
+                "operationId": "get-statistics",
+                "summary": "Get the statistics for a certain location",
+                "description": "# Purpose\nProvides the statistics for a location\n One can specify the period for which the statistics can be calculated. Als the purpose can be provided to limit the purpose of the statistics",
```
Als -> Also

AndreasSchultzGEA commented on url-schemes/managementURLScheme.json:
```diff
@@ -442,6 +442,48 @@
+                        "$ref": "#/components/parameters/location-id"
+                    },
+                    {
+                        "$ref": "#/components/parameters/date-from"
```
In the StatisticsResource, the date-properties are called start- and endDate. 
As date-from and dato-to act somehow as a filter, shouldn't they be equal?
And: are they sth. like a filter on event-dates? Or are they more general than gained from AnimalEvents?

AndreasSchultzGEA commented on types/icarGroupType.json:
```diff
@@ -0,0 +1,22 @@
+  "type": "object",
+
+  "properties": {
+    "icarGroupType": {
```
will lead to a icarGroupType in a icarGroupType. --> confusing

AndreasSchultzGEA commented on resources/icarStatisticsResource.json:
```diff
@@ -0,0 +1,41 @@
+      "$ref": "../types/icarLocationIdentifierType.json",
+      "description": "Unique location scheme and identifier combination."
+    },
+    "purpose": {
```
This is a repetition of what has been provided by the requester, but it is okay, I think.

AndreasSchultzGEA commented on types/icarGroupType.json:
```diff
@@ -0,0 +1,22 @@
+      "nullable": true,
+      "description": "Number of animals in the group."
+    },
+    "icarGroupSpecifier": {
```
The content of the icarGroupSpecifierType seem a little bit to be an own small statistic. I'm not sure if this information is always needed.
But: Why not inline the content of icarGroupSpecifierType in icarGrouptype instead of referencing it as a different type?

AndreasSchultzGEA commented on types/icarGroupSpecifierType.json:
```diff
@@ -0,0 +1,32 @@
+      "nullable": true,
+      "description": "maximum number of days in milk for the animals in the group."
+    },
+    "animalSetId": { 
```
Is an animalSetId always available? If you think of the enum/icarGroupType Herd, there won't be such id available

AndreasSchultzGEA commented on types/icarStatisticsType.json:
```diff
@@ -0,0 +1,23 @@
+
+  "properties": {
+    "metric": {
+      "$ref": "../types/icarMetricType.json"
```
I'm not sure if I understand what content could be provided within the metric?
What id and scheme should be used?

btw., why don't use 'icarIdentifierType' directly instead?

AndreasSchultzGEA commented on types/icarMetricType.json:
```diff
@@ -0,0 +1,7 @@
+{
+  "description": "Identifies a metric using a scheme and ID. Allows country or species-specific metrics that are a superset of the ICAR list.",
```
Is this type necessary at all?

AndreasSchultzGEA commented on types/icarStatisticsType.json:
```diff
@@ -0,0 +1,23 @@
+  "description": "The statitics that have been calculated.",
+
+  "type": "object",
+
```
no required properties?

AndreasSchultzGEA commented on resources/icarStatisticsResource.json:
```diff
@@ -0,0 +1,41 @@
+  "type": "object",
+
+  "required": [
+    "id"
```
Only id? What about purpose.
I'm not sure if group or statistics must also be present?
I think, if they miss, the complete Statistics-Resource may not be delivered?

In general: You may have a look at the required properties.

erwinspeybroeck committed:
Update icarStatisticsType.json

it seems statistics is not being rendered. Try to find out why ...

erwinspeybroeck committed:
Update icarStatisticsType.json

value ok, now added aggregation

erwinspeybroeck committed:
Update icarStatisticsType.json

added the rest, probably a missing comma ...

erwinspeybroeck committed:
Update icarStatisticsPurposeType.json

erwinspeybroeck committed:
Create icarStatisticsGroupType.json

Created to be used instead of icarGroupType.json
icarGroupType.json was used also in the enums section ...

erwinspeybroeck committed:
Update icarStatisticsResource.json

use of icarStatisticsGroupType instead of icarGroupType

erwinspeybroeck committed:
Delete icarGroupType.json

has been interchanged with icarStatisticsGroupType

erwinspeybroeck committed:
Update managementURLScheme.json

typo

erwinspeybroeck committed:
Update icarStatisticsResource.json

made all fields required. One needs to know the location,  purpose, the period, eventual group info and statistics

erwinspeybroeck committed:
Update icarStatisticsResource.json

equal to the filter parameters

erwinspeybroeck committed:
Update managementURLScheme.json

added some documentation

erwinspeybroeck committed:
Update icarGroupSpecifierType.json

added some documentation
